### DISCUSSION TITLE
Hidden ListItem in ListView is selected when navigating with keyboard

### DISCUSSION THREAD

**Discussion by paulrobello** [Textualize/textual#4261] [Q&A] [answered]
When a ListView contains ListItems with display = False the items are still selected when navigating ListView with keyboard.

[MRE](https://gist.github.com/paulrobello/5ba0d5a636fbef0a9ff52afeabd43d28)

**Comment by davep** [comment]
As it stands, `ListView` wasn't and isn't designed for this sort of use. It's there as a straightforward method of navigating a vertical list of `ListItem`s, it doesn't aim to be a full-featured container that will handle items that are contained but not displayed.

Moreover, even if it did, the code in the MRE wouldn't do the job anyway as you're hiding the `Label` inside the `ListItem`, and it's `ListItem`s that `ListView` handles, navigates and tracks.

It is worth noting, however, that `ListView` *will* handle a `ListItem` being `disabled`.

As an unrelated PS:

```python
        with Footer():
            yield Label("Use arrow keys to navigate. Hidden items are still selected")
```

this isn't really the intended use for `Footer`. While it will work,  it works *against* what the `Footer` widget is intended for (showing bindings within the context of the current focus chain).

**Comment by paulrobello** [comment]
I have updated the MRE to correctly hide the list item and not the label.
The DOM has a property displayed_children which seems like it should be the list of children that are used when navigating?
I suppose I could disable and hide the items to get the desired functionality.

**Reply by davep** [reply, marked as answer]
As I mentioned above: to best of my knowledge `ListItem` wasn't designed as a widget that could have hidden content; it isn't intended as a general container[^1].

Disable and hide seems like a reasonable approach at the moment.

[^1]: That's not to suggest that it shouldn't be changed to do so, I'm just pointing out how it works at the moment.
